champions/src/main/java/me/mykindos/betterpvp/champions/weapons/impl/legendaries/RunedPickaxe.java

```diff
@@ -0,0 +1,124 @@
+package me.mykindos.betterpvp.champions.weapons.impl.legendaries;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import me.mykindos.betterpvp.champions.Champions;
+import me.mykindos.betterpvp.core.client.repository.ClientManager;
+import me.mykindos.betterpvp.core.combat.weapon.types.ChannelWeapon;
+import me.mykindos.betterpvp.core.combat.weapon.types.InteractWeapon;
+import me.mykindos.betterpvp.core.combat.weapon.types.LegendaryWeapon;
+import me.mykindos.betterpvp.core.energy.EnergyHandler;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.Sound;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.block.BlockBreakEvent;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.meta.ItemMeta;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Singleton
+@BPvPListener
+public class RunedPickaxe extends ChannelWeapon implements InteractWeapon, LegendaryWeapon, Listener {
+
+    private double velocityStrength;
+    private final EnergyHandler energyHandler;
+    private final ClientManager clientManager;
+    private boolean cooldownActive;
```

Comment:
This is a global cooldown for every single player, it would have worked fine when you were the only person using it though.

You probably want to create a HashSet that just stores the unique ID of players that are currently on cooldown, and remove them if they meet the conditions